packages/core-data/src/actions.js

```diff
@@ -139,6 +139,52 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string} kind              Kind of the deleted entity.
+ * @param {string} name              Name of the deleted entity.
+ * @param {string} recordId          Record Id of the entity to be deleted.
+ * @param {string} query Special query parameters for the DELETE API call. These parameters will be appended to the rest base with a '?' in front.
```

Comment:
The query should be an object like in all the other methods. We have utilities to create the query strings.